src/http.cpp

```diff
@@ -684,21 +665,20 @@ m_off_t SpeedController::calculateSpeed(long long numBytes)
 
     while (transferBytes.size())
```

Comment:
I think that it should be more efficient to get an iterator to the beginning of the list on each iteration than calling erase with a key, that requires to find the element in the whole map.